### DISCUSSION TITLE
How to get or count the current index while "sliding" in Fullscreen mode?

### DISCUSSION THREAD

**Discussion by Phoi-Hoa-Nguyen** [igordanchenko/yet-another-react-lightbox#68] [Q&A] [answered]
Hey!

Thanks for the awesome library. But now I´m stuck at the point where how to code the proper function of my custom button in the Lightbox/Fullscreen/Slideshow mode. 
I could add the custom button within the Lightbox tag with following code:

                toolbar={{
                    buttons: [
                        <div key={0} className="yarl__custom_toolbar">
                            {index !== -1 && (
                                <ClipboardCopy copyText={slides[index].description} />
                            )}
                        </div>,
                        "close"
                    ]
                }}

It allows me to "copy" the current 'description' of the Image (if I´m outside the Fullscreen/Slideshow mode and choose one of the Image from the Photoalbum). But inside the Fullscreen/Slideshow mode while sliding to other images it doesn´t update the index. I don´t know the proper way to update the current index for the right Image + description for the "copy Button". Is it somehow possible or a way for it?

Thanks in advance!

**Comment by igordanchenko** [comment, marked as answer]
Hi there! I'm glad to hear you are enjoying this library so far. 

I can think of 2 approaches to access the current slide index in this scenario.

1) Use the `on.view` callback and store the current slide index in a ref.

```js
const slideIndex = React.useRef();

// ...

<Lightbox
  // ...
  on={{
    view: (index) => {
      slideIndex.current = index;
    }
  }}
/>
```

https://codesandbox.io/s/yet-another-react-lightbox-68-1-bjmv07?file=/src/App.js

2) Use the internal `useLightboxState` hook. Just keep in mind that you have to declare a component in order to use this hook, as it won't work in an inline render function.

```jsx
import { useLightboxState } from "yet-another-react-lightbox/core";

const CopyButton = () => {
  const {
    state: { currentIndex }
  } = useLightboxState();

  return (
    <button
      type="button"
      className="yarl__button"
      onClick={() => {
        console.log("Current slide index: " + currentIndex);
      }}
    >
      Copy
    </button>
  );
};

// ...

<Lightbox
  // ...
  toolbar={{ buttons: [<CopyButton key="copy" />, "close"] }}
/>
```

https://codesandbox.io/s/yet-another-react-lightbox-68-2-kvclwt?file=/src/App.js

I hope this helps!

**Reply by Phoi-Hoa-Nguyen** [reply]
Hey igordanchenko,
i tried the second solution with the internal useLightboxState hook. And it works like a charm!!! Thank you really much mate!!! 
And also for the fast reply and solutions. I appreciate it very much! :)
